Sources/TSCBasic/PathShims.swift

```diff
@@ -168,18 +168,20 @@ extension AbsolutePath {
     /// Returns a path suitable for display to the user (if possible, it is made
     /// to be relative to the current working directory).
     public func prettyPath(cwd: AbsolutePath? = localFileSystem.currentWorkingDirectory) -> String {
-        guard let dir = cwd else {
-            // No current directory, display as is.
+        guard let dir = cwd,
+            let rel = try? relative(to: dir) else {
+            // Cannot create relative path, display as is.
             return self.pathString
         }
-        // FIXME: Instead of string prefix comparison we should add a proper API
-        // to AbsolutePath to determine ancestry.
-        if self == dir {
-            return "."
-        } else if self.pathString.hasPrefix(dir.pathString + "/") {
-            return "./" + self.relative(to: dir).pathString
+        if let first = rel.components.first,
+            first != ".." {
```

Comment:
What is the situation you're checking here? `..`s should be removed as part of lexical normalization except at the start of relative paths.

Reply:
I just copied the old behavior here. The old code shows that the returned string starts with either `..` or `.`.